src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs

```diff
@@ -100,29 +101,38 @@ public static TypedData ToRpc(this object value)
                     object body = null;
                     string rawBody = null;
 
-                    switch (request.ContentType)
+                    MediaTypeHeaderValue mediaType = null;
+                    MediaTypeHeaderValue.TryParse(request.ContentType, out mediaType);
+
+                    if (string.Compare(mediaType.MediaType, "application/json", StringComparison.OrdinalIgnoreCase) == 0)
                     {
-                        case "application/json":
-                            var jsonReader = new StreamReader(request.Body, Encoding.UTF8);
-                            rawBody = jsonReader.ReadToEnd();
+                        var jsonReader = new StreamReader(request.Body, Encoding.UTF8);
+                        rawBody = jsonReader.ReadToEnd();
+                        try
+                        {
                             body = JsonConvert.DeserializeObject(rawBody);
-                            break;
-
-                        case "application/octet-stream":
-                        case string contentType when contentType.IndexOf("multipart/", StringComparison.OrdinalIgnoreCase) >= 0:
-                            var length = Convert.ToInt32(request.ContentLength);
-                            var bytes = new byte[length];
-                            request.Body.Read(bytes, 0, length);
-                            body = bytes;
-                            rawBody = Encoding.UTF8.GetString(bytes);
-                            break;
-                        default:
-                            var reader = new StreamReader(request.Body, Encoding.UTF8);
-                            body = rawBody = reader.ReadToEnd();
-                            break;
+                        }
+                        catch (JsonException)
```

Comment:
This is the actual fix. The rest is improvements to parsing, using MediaTypeHeaderValue

Reply:
Glad you created the other issue, the other code paths in here will continue to cause problems until that is addressed.